reductstore/src/storage/query/condition/parser.rs

```diff
@@ -99,24 +72,34 @@ impl Parser {
                                 Value::String(value.to_string())
                             }
                         }
-                        _ => {
-                            return Err(unprocessable_entity!(
-                                "Directive '{}' must be a primitive value",
-                                key
-                            ));
-                        }
-                    };
-                    vec![parsed]
+                        _ => panic!("Unsupported directive value type: {}", v), // panic because it's programming error
+                    }
                 };
 
-                directives.insert(key.to_string(), values);
+                let mut parsed_values = vec![];
+
+                if value.is_null() {
+                    return Err(unprocessable_entity!("Directive '{}' cannot be null", key));
+                } else if value.is_object() {
+                    return Err(unprocessable_entity!(
+                        "Directive '{}' cannot be an object",
+                        key
+                    ));
+                } else if value.is_array() {
+                    for item in value.as_array().unwrap() {
+                        parsed_values.push(parse_primitive(item));
+                    }
+                } else {
+                    parsed_values.push(parse_primitive(value))
+                }
+                directives.insert(key.to_string(), parsed_values);
                 keys_to_remove.push(key.to_string());
             }
         }
 
         // Remove directives from the original JSON object
         for key in keys_to_remove {
-            json.as_object_mut().unwrap().remove(&key);
+            json.as_object_mut().unwrap().shift_remove(&key);
```

Comment:
@AnthonyCvn FIY, this part broke the order. If a query had a directive, this method was called and it shuffled the operators in the query.